Group1/src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,159 @@
+import com.stuypulse.stuylib.control.feedback.PIDController;
+import com.stuypulse.stuylib.control.feedforward.Feedforward;
+import com.stuypulse.stuylib.network.SmartNumber;
+
+import static com.stuypulse.robot.constants.Ports.Drivetrain.Grayhill.*;
+import static com.stuypulse.robot.constants.Settings.Drivetrain.PID.*;
+import static com.stuypulse.robot.constants.Settings.Drivetrain.FF.*;
+
+/**
+ * Drivetrain fields:
+ * 
+ * Control Group:
+ * - FeedForward
+ * - PIDController
+ * - Kinematics
+ * 
+ * Hardware stuff:
+ * - Motor controller group (two)
+ * - Gyroscope (navx)
+ * - Encoders (grayhills)
+ * 
+ * Drivetrain methods:
+ * - arcadeDrive()
+ * - tankDrive()
+ * - reset()
+ * - getAngle()
+ */
+
+public class Drivetrain extends SubsystemBase {
+    /** CONTROLS */
+    private final DifferentialDriveKinematics driveKinematics;
+    private final Controller leftController;
+    private final Controller rightController;
+
+    private SmartNumber leftTargetSpeed;
+    private SmartNumber rightTargetSpeed;
+
+    /** HARDWARE */
+    private final MotorControllerGroup left;
+    private final MotorControllerGroup right;
+
+    private final Encoder leftGrayhill;
+    private final Encoder rightGrayhill;
+
+    private final AHRS navx;
+
+    /** ODOMETRY */
+    private final DifferentialDriveOdometry odometry;
+
+    private final Field2d field;
+
+    public Drivetrain() {
+        setSubsystem("Drivetrain");
+        /** CONTROL */
+        leftController = new PIDController(kP, kI, kD).add(new Feedforward.Drivetrain(kS, kV, kA).velocity());
+        rightController = new PIDController(kP, kI, kD).add(new Feedforward.Drivetrain(kS, kV, kA).velocity());
+
+        driveKinematics = new DifferentialDriveKinematics(Settings.Drivetrain.TRACK_WIDTH);
+
+        leftTargetSpeed = new SmartNumber("Drivetrain/Left Target Speed", 0.0);
+        rightTargetSpeed = new SmartNumber("Drivetrain/Right Target Speed", 0.0);
+
+        /** MOTORS */
+        CANSparkMax leftFront = new CANSparkMax(Ports.Drivetrain.LEFT_FRONT, MotorType.kBrushed);
+        CANSparkMax leftMiddle = new CANSparkMax(Ports.Drivetrain.LEFT_MIDDLE, MotorType.kBrushless);
+        CANSparkMax leftBottom = new CANSparkMax(Ports.Drivetrain.LEFT_BOTTOM, MotorType.kBrushless);
+        Motors.Drivetrain.left.configure(leftFront, leftMiddle, leftBottom);
+        left = new MotorControllerGroup(leftFront, leftMiddle, leftBottom);
+        addChild("Left Motor Controller Group", left);
+
+        CANSparkMax rightFront = new CANSparkMax(Ports.Drivetrain.RIGHT_FRONT, MotorType.kBrushless);
+        CANSparkMax rightMiddle = new CANSparkMax(Ports.Drivetrain.RIGHT_MIDDLE, MotorType.kBrushless);
+        CANSparkMax rightBottom = new CANSparkMax(Ports.Drivetrain.RIGHT_BOTTOM, MotorType.kBrushless);
+        Motors.Drivetrain.right.configure(rightFront, rightMiddle, rightBottom);
+        right = new MotorControllerGroup(rightFront, rightMiddle, rightBottom);
```

Comment:
change front to top

Reply:
front is good, bottom is not

front/back vs. top/bottom